jax2onnx/converter/ir_postprocess.py

```diff
@@ -0,0 +1,275 @@
+# jax2onnx/converter/ir_postprocess.py
+
+from __future__ import annotations
+
+from collections.abc import Iterable as IterableABC
+from typing import Iterable, Sequence, Any
+
+import numpy as np
+import onnx_ir as ir
+from onnx_ir import AttributeType as IRAttrType
+
+
+def _list_from_maybe_iterable(obj: object) -> list[Any]:
+    if obj is None:
+        return []
+    if isinstance(obj, (str, bytes)):
+        return []
+    if isinstance(obj, IterableABC):
+        try:
+            return list(obj)
+        except Exception:
+            return []
+    return []
+
+
+def _value_name(value: object) -> str | None:
+    if isinstance(value, str):
+        return value or None
+    return getattr(value, "name", None)
+
+
+def _shape_dims(shape_obj: object) -> Sequence[object] | None:
+    if shape_obj is None:
+        return None
+    dims = getattr(shape_obj, "dims", None)
+    if dims is not None:
+        return list(dims)
+    if isinstance(shape_obj, (list, tuple)):
+        return list(shape_obj)
+    return None
+
+
+def _dim_is_known(dim: object) -> bool:
+    if dim is None:
+        return False
+    if isinstance(dim, (int, np.integer)):
+        return True
+    if isinstance(dim, str):
+        return bool(dim)
+    for attr in ("param", "name", "symbol", "label"):
+        val = getattr(dim, attr, None)
+        if val:
+            return True
+    value_attr = getattr(dim, "value", None)
+    if isinstance(value_attr, (int, np.integer)):
+        return True
+    try:
+        text = str(dim)
+        if text and text.isidentifier():
+            return True
+        if "SymbolicDim" in text:
+            return True
+    except Exception:
+        pass
+    return False
```

Comment:
This is overly complex: call https://onnx.ai/ir-py/api/generated/onnx_ir.Shape.html#onnx_ir.Shape.is_unknown_dim